cognoml/analysis.py

```diff
@@ -0,0 +1,103 @@
+import collections
+#import os
+import warnings
+
+import pandas as pd
+import numpy as np
+import sklearn
+from sklearn.cross_validation import train_test_split
+
+from cognoml import utils
+from cognoml.classifiers.logistic_regression import clf_grid, pipeline
+
+# expression_path = os.path.join('download', 'mutation-matrix.tsv.bz2')
+expression_path = "https://github.com/cognoma/cancer-data/raw/54140cf6addc48260c9723213c40b628d7c861da/data/subset/expression-matrix-all-samples.tsv"
+
+def read_data():
+    """
+    Read data.
+    """
+    # Read expression data
+    X = pd.read_table(expression_path, index_col=0)
+    return X
+
+def classify(sample_id, mutation_status, **kwargs):
+    """
+    Perform an analysis.
+    
+    Parameters
+    ----------
+    sample_id : list
+        Sample IDs of the observations.
+    mutation_status : list
+        Mutation status (0 or 1) of each sample.
+    """
+    results = collections.OrderedDict()
+    
+    obs_df = pd.DataFrame.from_items([
+        ('sample_id', sample_id),
+        ('status', mutation_status),
+    ])
+    
+    X_whole = read_data()
+    X = X_whole.loc[obs_df.sample_id, :]
+    y = obs_df.status
+    
+    X_train, X_test, y_train, y_test = train_test_split(
```

Comment:
also, I was at talk by Olivier Elemento - he was building models for a different purpose (predict immunotherapy responders) but was adjusting for mutation burden as a covariate. We may want to consider checking out his stuff and adjusting for burden too